File: react-type-vite/src/services/api/response/quizResponse.ts
import type { QuestionResponse } from "./questionResponse"

interface QuizResponse {
  id: number
  sectionId: number
  sectionName: string
  title: string
  description: string
  duration: number
  attemptLimit: number
  passingScore: number
  numberItem: number
  showResults: boolean
  isPublished: boolean
  startTime?: string;
  endTime?: string;
  questions: Set<QuestionResponse>
  blueprintDraft?: { cloId: number; percentage: number }[]
  attemptsCount: number
  createdAt: Date
  updateAt: Date
}

export type { QuizResponse };

File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/QuizResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuizResponse {

    private Integer id;

    private Integer sectionId;

    private String sectionName;

    private String title;

    private String description;

    private Integer duration;

    private Integer attemptLimit;

    private Double passingScore;

    private Integer numberItem;

    private Boolean showResults;

    private Boolean isPublished;

    private Set<QuestionResponse> questions;

    private Integer attemptsCount;

    private LocalDateTime startTime;

    private LocalDateTime endTime;

    private Date createdAt;

    private Date updateAt;
}
